src/tools/illink/src/ILLink.Tasks/build/Microsoft.NET.ILLink.targets

```diff
@@ -196,9 +196,8 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- The defaults currently root non-framework assemblies, which
          is a no-op for portable apps. If we later support more ways
          to customize the behavior we can allow linking portable apps
-         in some cases. If we're not running ILLink because e.g. this
-         is a NativeAOT app, value of SelfContained doesn't matter. -->
-    <NETSdkError Condition="'$(RunILLink)' != 'false' And '$(SelfContained)' != 'true'" ResourceName="ILLinkNotSupportedError" />
+         in some cases. -->
+    <NETSdkError Condition="'$(SelfContained)' != 'true'" ResourceName="ILLinkNotSupportedError" />
```

Comment:
If we insist on making this change, the error message could use some work for this new scenario. Currently it reads:

> Optimizing assemblies for size is not supported for the selected publish configuration

It may not be readily apparent to users that  "Optimizing assemblies for size" means `PublishAot` in this case.